### ISSUE TITLE
Frequent crashes when using HiPS layers

### ISSUE THREAD

**Issue by Atque** [Stellarium/stellarium#829]
I have noticed that Stellarium often crashes when using the HiPS layers (and sometimes also using the TOAST survey, but more rarely). Especially if you show a planet in view, zoom in on it, Stellarium crashes.

### Steps to reproduce
1. Open HiPS layer (DSS color is nice!)
2. Zoom in and center on the moon.
3. Now scroll through dates.
4. After a while, Stellarium will crash.

If you don't succeed in creating a crash, try zooming in a out on planets and moons. Also use the ocular.

### System
* Stellarium version: stellarium-0.19.2.17327-win64.exe (latest public beta, but this problem is old)
* Operating system: Windows 10 64-bit
* Graphics Card: GeForce GTX 1070 with latest drivers.

### Logfile 
This is a logfile from such a crash, if it tells anything:

[log.txt](https://github.com/Stellarium/stellarium/files/3796746/log.txt)

**Comment by alex-w** [comment]
Cannot confirm.

**Comment by gzotti** [comment]
The only times I had crashes with TOAST and HiPS was on my Raspberry Pi 3 when it presumably ran out of texture memory or had other issues with its still not perfect GPU drivers. (Have not checked with RPi4 recently.) But on a 1070? I had overnight runs on a 580M to give a counter example.

**Comment by Atque** [comment]
@gzotti Running out of memory was my thought too, but it seems unlikely for this card. I tried a fresh installation with a fresh config, no additional star catalogs etc, but still the same problem. :/

It happens when I do too jerky pannings/zooms. It often happens with HiPS and very rarely with TOAST.

**Comment by gzotti** [comment]
OK, jerky panning may be a hint. @guillaumechereau any ideas?

**Comment by gzotti** [comment]
I am not able to break it, be it with jerky movement, fast zoom or other ways. I see dynamic reloading, but no crash. Does it depend on particular HiPS? Maybe the logfile says anything?

**Comment by Atque** [comment]
@gzotti This is a logfile from a clean installation (no user data, everything set to default).
[log.txt](https://github.com/Stellarium/stellarium/files/3801331/log.txt)

To make it crash, I displayed the **DSS colored** HiPS survey, disabled atmosphere and ground, zoomed in on the Moon (about 1-2 degrees FOV), centered it and scrolled through dates. After about 10-15 seconds, Stellarium froze for a split second, then closed down. Again, this was on a clean installation (no additional star catalogs, default config.ini, no user locations etc).

Same problem with the Mellinger survey. However, I cannot create a crash with the blue and red bands of the DSS. I fooled around with the moon for about one minute without a crash. 

I remember this particular commit made crashes occur slightly less frequently:
https://github.com/Stellarium/stellarium/commit/fba9826497609e36411bdf098be020526f6aa950

**Comment by alex-w** [comment]
@Atque please give us URL of DSS source

**Comment by Atque** [comment]
@alex-w http://alaskybis.u-strasbg.fr/DSS/DSSColor

**Comment by Atque** [comment]
After getting a similar crash without displaying either HiPS or TOAST, I'm starting to think this is not an HiPS problem after all.

I do not get the problems when using ANGLE or MESA modes. OpenGL issue?

**Comment by gzotti** [comment]
Hard to say. I am still not able to break it, and your logfile does not say "I am going to crash now because ...". :-) I am running drivers 436.15, maybe try that?

**Comment by Atque** [comment]
@gzotti I tried downloading and installing the Studio driver of my card, instead of the Game Ready driver. Now it seems a lot more stable (at least I haven't managed to create a crash).

**Comment by Atque** [comment]
I guess we can close this now. It seems to have been caused by a driver problem out of Stellarium's team's reach.

**Comment by axd1967** [comment]
I suspect that this issue might still be valid, because I have the impression that it can happen when playing a lot with HiPS. I'll come back here and if needed, open an issue.

**Comment by Atque** [comment]
@axd1967 The computer I was using when I reported this bug had frequent crashes with the HiPS and planet/moon in frame. With my new computer, with an RX 5700 XT, I have not had this problem.

On my old computer with my old card, I tried everything. Different drivers and even reinstalled Windows 10. Still no success. The only thing that prevented a crash was using Stellarium in ANGLE mode. So this must have been some bug with OpenGL and that particular card (GeForce GTX 1070). It would be interesting to see if other people with the same card have these problems.
